research/conditional/utils/argparse.py

```diff
@@ -199,6 +199,11 @@ def introduce_parser_arguments(
         action="store_true",
         help="in grouped ExpertChoice, use squash all linears with einsum",
     )
+    parser.add_argument(
+        "--use_torch_bmm",
+        action="store_true",
+        help="in grouped ExpertChoice, use full_einsum implementation with all einsums replaced by torch.bmm",
```

Comment:
would this be able to not reference `full_einsum`? This would make the description clearer
I assume it means it is a fully linear implementation (without index_add), so maybe that?

Reply:
Ok, I'll just change it to "one hot implementation"